airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1779,6 +1779,10 @@ def execute(self, context: Context):
         self.log.info("Template instantiated. Workflow Id : %s", self.workflow_id)
         operation.result()
         self.log.info("Workflow %s completed successfully", self.workflow_id)
+    
+    def on_kill(self):
+        if self.operation:
```

Comment:
You'll need to make `operation` an instance attr in `execute()` for this method to work too.